oolab/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -0,0 +1,44 @@
+package agh.ics.oop;
+
+import java.util.Vector;
+
+import static agh.ics.oop.MapDirection.*;
+
+public class Animal {
+    private MapDirection orientation;
+    private Vector2d position;
+    public Animal(){
+        orientation = MapDirection.NORTH;
+        position = new Vector2d(2,2);
+    }
+
+    public MapDirection getOrientation(){
+        return orientation;
+    }
+
+    @Override
+    public String toString(){
+        return ("Location: " + position.toString() + "\nOrientation: " + orientation);
+    }
+
+    public boolean isAt(Vector2d position){
+        return position.equals(this.position);
+    }
+
+    public void move(MoveDirection direction){
+        if((this.position.x() == 0 && direction == MoveDirection.LEFT) || (this.position.x() == 4 && direction == MoveDirection.RIGHT) || (this.position.y() == 0 && direction == MoveDirection.BACKWARD) || (this.position.y() == 4 && direction == MoveDirection.FORWARD))
+            return;
+        switch (direction){
```

Comment:
w tym i kolejnym switchu trochę wymyślanie koła na nowo - mamy już do tego metody typu `toUnitVector()`, `previous()`, `next()` itp.

Reply:
Jeśli nie posiadam żadnego błędu, to klasa MoveDirection nie posiada żadnych metod. Stąd takie obchodzenie tego. Mogę je dopisać i wykorzystać tutaj, jednakże wydaje mi się, że nie jest to zgodne z poleceniami.

Reply:
Okazuje się, że kompletnie źle zrozumiałem zadanie. Teraz jest dobrze